plugins/omelasticsearch/omelasticsearch.c

```diff
@@ -1084,14 +1151,14 @@ static rsRetVal
 curlPost(wrkrInstanceData_t *pWrkrData, uchar *message, int msglen, uchar **tpls, int nmsgs)
 {
 	CURLcode code;
-	CURL *curl = pWrkrData->curlHandle;
+	CURL *curl = pWrkrData->curlPostHandle;
 	DEFiRet;
 
 	pWrkrData->reply = NULL;
 	pWrkrData->replyLen = 0;
 
-	if(pWrkrData->pData->dynSrchIdx || pWrkrData->pData->dynSrchType || pWrkrData->pData->dynParent)
-		CHKiRet(setCurlURL(pWrkrData, pWrkrData->pData, tpls));
+	CHKiRet(checkConn(pWrkrData));
```

Comment:
I went back and forth on whether this should be done only for bulk mode, or all the time, and finally settled on all the time since in either case, the ratio for the number of requests made is still ~2x.

By opting out of bulk mode, the user is already accepting significant overhead in the number of requests. Perhaps the default should be bulkmode=on?